diff --git a/src/Makefile.in b/src/Makefile.in
index f3d53be..92a536e 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -814,7 +814,7 @@ uninstall-am: uninstall-binSCRIPTS uninstall-dbusDATA \
 	  -e "s,::DEFAULT_QEMU_USER::,$(DEFAULT_QEMU_USER)," \
 	  -e "s,::ENABLE_UNSUPPORTED_RHEL_OPTS::,$(ENABLE_UNSUPPORTED_RHEL_OPTS)," \
 	  -e "s|::PREFERRED_DISTROS::|$(PREFERRED_DISTROS)|" \
-	  -e "s|::HV_PACKAGES::|$(KVM_PACKAGES)|" \
+	  -e "s|::KVM_PACKAGES::|$(KVM_PACKAGES)|" \
 	  -e "s|::ASKPASS_PACKAGE::|$(ASKPASS_PACKAGE)|" \
 	  -e "s|::LIBVIRT_PACKAGES::|$(LIBVIRT_PACKAGES)|" \
 	  < $< > $@
