diff -rup virt-manager-0.5.3.old/src/virt-manager.py.in virt-manager-0.5.3/src/virt-manager.py.in
--- virt-manager-0.5.3.old/src/virt-manager.py.in	2008-03-05 17:00:38.000000000 -0500
+++ virt-manager-0.5.3/src/virt-manager.py.in	2008-03-05 17:01:01.000000000 -0500
@@ -172,7 +172,7 @@ def show_engine(engine, show, uri, uuid)
         if engine.config.get_connections() is None or len(engine.config.get_connections()) == 0:
             tryuri = None
             if os.path.exists("/var/lib/xend") and os.path.exists("/proc/xen"):
-                tryuri = "xen"
+                tryuri = "xen:///"
             elif os.path.exists("/usr/bin/qemu"):
                 if os.getuid() == 0:
                     tryuri = "qemu:///system"
