diff -rup virt-manager-0.6.0/src/virtManager/connection.py new/src/virtManager/connection.py
--- virt-manager-0.6.0/src/virtManager/connection.py	2008-09-10 19:17:43.170355000 -0400
+++ new/src/virtManager/connection.py	2008-09-10 19:19:58.124618000 -0400
@@ -376,6 +376,9 @@ class vmmConnection(gobject.GObject):
         self.connectThread.start()
 
     def _do_creds_polkit(self, action):
+        if os.getuid() == 0:
+            logging.debug("Skipping policykit check as root")
+            return 0
         logging.debug("Doing policykit for %s" % action)
         bus = dbus.SessionBus()
 
