diff -rup virt-manager-0.6.1/src/vmm-details.glade new/src/vmm-details.glade
--- virt-manager-0.6.1/src/vmm-details.glade	2009-01-26 14:33:33.000000000 -0500
+++ new/src/vmm-details.glade	2009-03-04 12:31:51.000000000 -0500
@@ -218,15 +218,15 @@
                 <child>
                   <widget class="GtkMenu" id="details-menu-send-key_menu">
                     <child>
-                      <widget class="GtkMenuItem" id="details-menu-send-cad">
+                      <widget class="GtkMenuItem" id="details-menu-send-cab">
                         <property name="visible">True</property>
                         <property name="label" translatable="no">Ctrl+Alt+Backspace</property>
                         <property name="use_underline">True</property>
-                        <signal name="activate" handler="on_details_menu_send_cad_activate"/>
+                        <signal name="activate" handler="on_details_menu_send_cab_activate"/>
                       </widget>
                     </child>
                     <child>
-                      <widget class="GtkMenuItem" id="details-menu-send-cad1">
+                      <widget class="GtkMenuItem" id="details-menu-send-cad">
                         <property name="visible">True</property>
                         <property name="label" translatable="no">Ctrl+Alt+Delete</property>
                         <property name="use_underline">True</property>
