diff -rup virt-manager-0.5.2.orig/src/virtManager/console.py virt-manager-0.5.2.new/src/virtManager/console.py
--- virt-manager-0.5.2.orig/src/virtManager/console.py	2007-10-04 21:43:56.000000000 -0400
+++ virt-manager-0.5.2.new/src/virtManager/console.py	2007-10-15 16:36:55.000000000 -0400
@@ -435,7 +435,7 @@ class vmmConsole(gobject.GObject):
             if credList[i] == gtkvnc.CREDENTIAL_PASSWORD:
                 self.activate_auth_page()
             elif credList[i] == gtkvnc.CREDENTIAL_CLIENTNAME:
-                self.vncViewer.set_credential(credList[i], "libvirt")
+                self.vncViewer.set_credential(credList[i], "libvirt-vnc")
             else:
                 # Force it to stop re-trying
                 self.vncViewerRetriesScheduled = 10
