commit 4922222c2c7b13704b6f461aedbee9ec06c01b5b
Author: Cole Robinson <crobinso@redhat.com>
Date:   Tue Apr 5 16:15:05 2011 -0400

    addhw: Fix nonhotplug to running guest

diff --git a/src/virtManager/addhardware.py b/src/virtManager/addhardware.py
index 885c8b9..7fa5b6e 100644
--- a/src/virtManager/addhardware.py
+++ b/src/virtManager/addhardware.py
@@ -938,7 +938,8 @@ class vmmAddHardware(vmmGObjectUI):
                   "Would you like to make the device available after the "
                   "next VM shutdown?")),
                 dialog_type=gtk.MESSAGE_WARNING,
-                buttons=gtk.BUTTONS_YES_NO)
+                buttons=gtk.BUTTONS_YES_NO,
+                async=False)
 
             if not res:
                 return (False, None)
