--- virt-manager-0.8.7/configure	2011-03-24 16:41:26.000000000 -0400
+++ /home/crobinso/sandbox/upstream/virt-manager/virt-manager.git/configure	2011-03-29 09:27:40.115612001 -0400
@@ -718,7 +718,7 @@ with_libvirt_package_names
 with_kvm_packages
 enable_unsupported_rhel_options
 with_preferred_distros
-with___with_default_graphics
+with_default_graphics
 '
       ac_precious_vars='build_alias
 host_alias
@@ -6521,7 +6521,9 @@ fi
 
 # Check whether --enable-unsupported-rhel-options was given.
 if test "${enable_unsupported_rhel_options+set}" = set; then :
-  enableval=$enable_unsupported_rhel_options; test "x$enableval" == "xno" && ENABLE_UNSUPPORTED_RHEL_OPTS=0
+  enableval=$enable_unsupported_rhel_options; test "x$enableval" == "xno" && \
+   ENABLE_UNSUPPORTED_RHEL_OPTS=0 ||
+   ENABLE_UNSUPPORTED_RHEL_OPTS=1
 else
   ENABLE_UNSUPPORTED_RHEL_OPTS=1
 fi
@@ -6537,9 +6539,9 @@ fi
 
 
 
-# Check whether --with---with-default-graphics was given.
-if test "${with___with_default_graphics+set}" = set; then :
-  withval=$with___with_default_graphics; DEFAULT_GRAPHICS=$withval
+# Check whether --with-default-graphics was given.
+if test "${with_default_graphics+set}" = set; then :
+  withval=$with_default_graphics; DEFAULT_GRAPHICS=$withval
 else
   DEFAULT_GRAPHICS="vnc"
 fi
