diff -rup virt-manager-0.6.0/src/vmm-create.glade new/src/vmm-create.glade
--- virt-manager-0.6.0/src/vmm-create.glade	2008-09-10 15:37:12.000000000 -0400
+++ new/src/vmm-create.glade	2008-11-25 12:08:47.161923000 -0500
@@ -871,7 +871,7 @@
 			  <child>
 			    <widget class="GtkLabel" id="label296">
 			      <property name="visible">True</property>
-			      <property name="label" translatable="yes">Lightweight method of virtualizing machines. Limits operating system choices because the OS must be specially modified to support paravirtualization. Better performance than fully virtualized systems.</property>
+			      <property name="label" translatable="yes">Lightweight method of virtualizing machines. Limits operating system choices because the OS must be specially modified to support paravirtualization, but yeilds better performance than fully virtualized systems.</property>
 			      <property name="use_underline">False</property>
 			      <property name="use_markup">True</property>
 			      <property name="justify">GTK_JUSTIFY_LEFT</property>
@@ -949,7 +949,7 @@
 				  <child>
 				    <widget class="GtkLabel" id="label297">
 				      <property name="visible">True</property>
-				      <property name="label" translatable="yes">Involves hardware simulation, allowing for a greater range of operating systems (does not require OS modification). Slower than paravirtualized systems.</property>
+				      <property name="label" translatable="yes">Involves hardware simulation, allowing for a greater range of virtual devices and operating systems (does not require OS modification).</property>
 				      <property name="use_underline">False</property>
 				      <property name="use_markup">True</property>
 				      <property name="justify">GTK_JUSTIFY_LEFT</property>
