diff -rup new/src/virtManager/connection.py new2/src/virtManager/connection.py
--- new/src/virtManager/connection.py	2009-12-03 17:14:57.324966000 -0500
+++ new2/src/virtManager/connection.py	2009-12-03 17:16:34.904055000 -0500
@@ -170,7 +170,8 @@ class vmmConnection(gobject.GObject):
         """
         Determine how we will be polling for net devices (HAL or libvirt)
         """
-        if self.is_nodedev_capable() and self.interface_capable:
+        if (self.is_remote() and
+            self.is_nodedev_capable() and self.interface_capable):
             try:
                 self._build_libvirt_netdev_list()
                 self.netdev_use_libvirt = True
@@ -207,7 +208,7 @@ class vmmConnection(gobject.GObject):
                 logging.debug("Using HAL for netdev enumeration")
 
     def _init_optical(self):
-        if self.is_nodedev_capable():
+        if self.is_remote() and self.is_nodedev_capable():
             try:
                 self.connect("nodedev-added", self._nodedev_optical_added)
                 self.connect("nodedev-removed", self._nodedev_optical_removed)
